src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -1,67 +1,211 @@
 import com.google.api.services.dataproc.Dataproc
-import com.google.api.services.pubsub.PubsubScopes
-import org.broadinstitute.dsde.workbench.leonardo.model.{ClusterRequest, ClusterResponse}
+import com.google.api.services.plus.PlusScopes
+import com.google.api.services.storage.{Storage, StorageScopes}
+import com.google.api.services.storage.model.{Bucket, ComposeRequest, StorageObject}
+import org.broadinstitute.dsde.workbench.leonardo.model.{ClusterInitValues, ClusterRequest, ClusterResponse}
 import org.broadinstitute.dsde.workbench.model.{ErrorReport, WorkbenchExceptionWithErrorReport}
 import org.broadinstitute.dsde.workbench.leonardo.errorReportSource
+
 import scala.collection.JavaConverters._
 import scala.concurrent.{ExecutionContext, Future}
 
 
-class GoogleDataprocDAO(protected val dataprocConfig: DataprocConfig)(implicit val system: ActorSystem, val executionContext: ExecutionContext)
+class GoogleDataprocDAO(protected val dc: DataprocConfig)(implicit val system: ActorSystem, val executionContext: ExecutionContext)
   extends DataprocDAO with GoogleUtilities {
 
   private val httpTransport = GoogleNetHttpTransport.newTrustedTransport
   private val jsonFactory = JacksonFactory.getDefaultInstance
-  private val cloudPlatformScopes = List(PubsubScopes.CLOUD_PLATFORM)
+  private val cloudPlatformScopes = List(ComputeScopes.CLOUD_PLATFORM)  // Maybe this would not need to be so high if we used ComputeScopes instead of Pubsub
+  private val storageScopes = Seq(StorageScopes.DEVSTORAGE_FULL_CONTROL, ComputeScopes.COMPUTE, PlusScopes.USERINFO_EMAIL, PlusScopes.USERINFO_PROFILE)
+  private val vmScopes = List(ComputeScopes.COMPUTE, ComputeScopes.CLOUD_PLATFORM)
+  private val serviceAccountPemFile = new File(dc.configFolderPath + dc.serviceAccountPemName)
 
   private def getDataProcServiceAccountCredential: Credential = {
     new GoogleCredential.Builder()
       .setTransport(httpTransport)
       .setJsonFactory(jsonFactory)
-      .setServiceAccountId(dataprocConfig.serviceAccount)
+      .setServiceAccountId(dc.serviceAccount)
       .setServiceAccountScopes(cloudPlatformScopes.asJava)
-      .setServiceAccountPrivateKeyFromPemFile(new java.io.File(dataprocConfig.serviceAccountPemPath))
+      .setServiceAccountPrivateKeyFromPemFile(serviceAccountPemFile)
+      .build()
+  }
+
+  def getBucketServiceAccountCredential: Credential = {
+    new GoogleCredential.Builder()
+      .setTransport(httpTransport)
+      .setJsonFactory(jsonFactory)
+      .setServiceAccountId(dc.serviceAccount)
+      .setServiceAccountScopes(storageScopes) // grant bucket-creation powers
+      .setServiceAccountPrivateKeyFromPemFile(serviceAccountPemFile)
       .build()
   }
 
+  private def getVmServiceAccountCredential: Credential = {
+    new GoogleCredential.Builder()
+      .setTransport(httpTransport)
+      .setJsonFactory(jsonFactory)
+      .setServiceAccountId(dc.serviceAccount)
+      .setServiceAccountScopes(vmScopes.asJava)
```

Comment:
`storageScopes` doesn't need `asJava`?  Maybe because it's a `Seq()` - if that's true, we should make them all `Seq()`s

Reply:
Probably because we're importing `scala.collection.JavaConversions._` which is deprecated. I think we should use JavaConverters and put `.asJava` everywhere it's needed.